six/modules/c++/cphd/include/cphd/Metadata.h

```diff
@@ -71,8 +74,11 @@ struct Metadata : MetadataBase
     size_t getCompressedSignalSize(size_t channel) const override;
     bool isCompressed() const override;
 
-    std::string getUri() const;
-    void setUri(const std::string& version);
+    //! Get version for new header
```

Comment:
I would remove the second part of the comment, because writing the header isn't the only reason to ask for the version.